packages/runed/src/lib/utilities/Context/Context.ts

```diff
@@ -0,0 +1,72 @@
+import { getContext, hasContext, setContext } from "svelte";
+
+export class Context<TContext> {
+	readonly #name: string;
+	readonly #key: symbol;
+
+	/**
+	 * @param name The name of the context.
+	 * This is used for generating the context key and error messages.
+	 */
+	constructor(name: string) {
```

Comment:
Perhaps we consider accepting a symbol here as well and using `Symbol.for` in that case?

Reply:
The name is only there to generate a readable error message. There's no benefit to accepting a custom key as the entire point of the util is to handle keys for you.

I also think using Symbol.for could potentially cause conflicts because it would reuse an existing symbol. Name isn't supposed to be unique, it's just a readable description.